pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -520,6 +529,10 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 	// Thus, we have to tear down the forwarding rule if either it or the target
 	// pool needs to be updated.
 	if fwdRuleExists && (fwdRuleNeedsUpdate || tpNeedsUpdate) {
+		// If we have to delete the forwarding rule, and something should fail
+		// before we recreate it, don't release the IP.  That way we can come
+		// back to it later.
+		orphanStaticIP = true
```

Comment:
You may be able to hack around the problem by having createOrPromoteStaticIP return whether the static IP already existed, and if it did, setting orphanStaticIP to true.

This code is pretty convoluted to begin with, though, so I'd hope that it isn't our idea of a permanent fix, just a temporary band-aid.